include/real/exact_number.hpp

```diff
@@ -643,46 +635,122 @@ namespace boost {
                     x /= BASE;
                 }
             }
+        
+            // returns {integer_part, decimal_part, exponent, is_positive}
+            static std::tuple<std::string, std::string, exponent_t, bool> number_from_string(std::string number) {
+                std::string integer_part;
```

Comment:
Isn't string_view enough in these cases?

Reply:
I think I can change it to use string_view, but I'll have to remove usage of .push_back() and .erase(), which means I'll be using your suggestions to do everything in one iteration. Also, since it doesn't support operator+, I'll have to make copies to get the numerator string as in the real.hpp string constructor.